pyproject.toml

```diff
@@ -26,13 +26,7 @@ target-version = "py38"
 
 # NOTE: Synchoronize the ignores with .flake8
 ignore = [
-    # these ignores are from flake8-bugbear; please fix!
-    "B008", "B017",
```

Comment:
Everything B01 and above 7 should probably be fixed instead of ignored. I just added those ignores so that they could be fixed manually eventually when I updated bugbear.

Reply:
These are the rules where autofix is not available. I think it's more realistic to (1) find and remove the noqa when more auto fix is implemented in ruff and do a batch fix and/or (2) Allow code owners to fix cases by case when they see the noqa (as a more obvious reminder in code).

Reply:
> These are the rules where autofix is not available.

Yet, ruff is rapidly adding many autofixes and if you file an issue they will often add it for simple fixes (like B007). 

If we do this, we need to differentiate these noqa from actual intentional noqas with some additional text in the comment (pretty sure you can either prepend or append text to the noqa and it will still work, but I forgot which). A useful feature for having a type ignore and noqa on the same line.

If I recall, some of these error codes have 20 or less violations and should be fixed manually. Some of the more involved ones will take singificantly more work for sure.

Reply:
https://github.com/charliermarsh/ruff/issues/4163 B027 should be trivial to add.